File: src/goLang/easy/q349/Solution.go
// q349 #easy #Intersection of Two Arrays #go
func intersection(nums1 []int, nums2 []int) []int {
	set1 := make(map[int]struct{})
	for _, num := range nums1 {
		set1[num] = struct{}{}
	}

	set2 := make(map[int]struct{})
	for _, num := range nums2 {
		set2[num] = struct{}{}
	}

	resultSet := make(map[int]struct{})
	for num := range set1 {
		if _, found := set2[num]; found {
			resultSet[num] = struct{}{}
		}
	}

	result := make([]int, 0, len(resultSet))
	for num := range resultSet {
		result = append(result, num)
	}

	return result
}

File: src/main/java/easy/q349/Solution.java
package easy.q349;

import java.util.Arrays;
import java.util.Set;
import java.util.stream.Collectors;

// q349 #easy #Intersection of Two Arrays #java

public class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        Set<Integer> set1 = Arrays.stream(nums1).boxed().collect(Collectors.toSet());
        Set<Integer> set2 = Arrays.stream(nums2).boxed().collect(Collectors.toSet());
        set1.retainAll(set2);
        return set1.stream().mapToInt(Integer::intValue).toArray();
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        int[] nums1 = {1, 2, 2, 1};
        int[] nums2 = {2, 2};
        int[] result = solution.intersection(nums1, nums2);
        for (int i : result) {
            System.out.println(i);
        }
    }
}
